### PR TITLE
Fix render latency

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

- Fix one-frame latency when rendering a new instance (#6717).
- Improve startup CPU time by 5% (250ms, loading Orders on my dev box).

The bug had to do with the indirection that allows a `ShapeView` to change what `Sprite` it owns in order to support layer changes. Currently, the indirection is implemented through changes in the display object hierarchy. However, in this approach, we must alter display object relationships (`add_child`) in a handler for an event emitted during display object update (`on_layer_change`). This causes tight coupling between the event handler and the display object update logic, which led to the bug.

The solution here is to eliminate the display objects at the `ShapeInstance` and `Sprite` levels, and connect `ShapeInstance` parameters to the events of a display object owned by the `ShapeView`, such that the needed indirection happens through the event handlers accessing the model's current `shape`. This eliminates the latency, and saves 250ms of CPU time that was spent on display object operations.

I believe, optimization aside, this change also makes sense from a design perspective. We support operations on display object such as moving them between layers. A `Sprite` cannot be moved between layers--they exist at a lower level of abstraction than display objects.

### Important Notes

- A new `SpriteObject` supports use of `Sprite`s without `Shape`s, e.g. in example scenes.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] ~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] ~Unit tests have been written where possible.~
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by wdanilo** [comment]
I'll review it tomorrow. Sorry for that @kazcw. Due to my sickness, the amount of notifications accumulated, I was trying to go trough all the issues tonight, but its so late, I'm not able to go trough this change, as its pretty dense in some places. Tomorrow the review will be here.

File: lib/rust/ensogl/examples/dom-symbols/src/lib.rs [hunk 1]
```diff
@@ -83,17 +83,18 @@ pub fn main() {
...
     let navigator = Navigator::new(scene, &camera);
     let sprite_system = SpriteSystem::new("test_sprite_system", alignment::Dim2::center());
-    world.add_child(&sprite_system);
+    world.add_child(&sprite_system.symbol);
 
     let dom_front_layer = &scene.dom.layers.front;
     let dom_back_layer = &scene.dom.layers.back;
 
-    let mut sprites: Vec<Sprite> = default();
+    let mut sprites: Vec<SpriteObject> = default();
```
Comment (wdanilo) [inline lib/rust/ensogl/examples/dom-symbols/src/lib.rs]: Can we rename `SpiteObject` to `Sprite` and `Sprite` to something else? Sprites by definitions are things displayed on the screen, so I thing the name of just "Sprite" will be better for the user.

**Review by wdanilo — APPROVED** [review]

File: lib/rust/ensogl/examples/dom-symbols/src/lib.rs (hunk 1 above)
Comment (kazcw) [reply lib/rust/ensogl/examples/dom-symbols/src/lib.rs]: I will rename the lower-level thing `RawSprite`; it is used internally by sprite implementations that complete it by associating it with a display object (in different ways).

**Comment by kazcw** [comment]
I will merge this into #6707 (now that they have both completed code review), so that they can be QA'd together. They both make changes that require fairly broad testing.

**Comment by kazcw** [comment]
I will merge this into https://github.com/enso-org/enso/pull/6707 (now that they have both completed code review), so that they can be QA'd together. They both make changes that require fairly broad testing.

**Pull request closed without merging**